includes/functions.php

```diff
@@ -4611,3 +4611,89 @@ function dokan_override_author_for_product_variations( $product, $seller_id ) {
         }
     }
 }
+
+/**
+ * Handle user update from customer to seller.
+ *
+ * @since DOKAN_SINCE
+ *
+ * @param object $user User Object
+ * @param array  $data Data to Update
+ *
+ * @return void
+ */
+if ( ! function_exists( 'dokan_user_update_to_seller' ) ) {
+
+    function dokan_user_update_to_seller( $user, $data ) {
+        if ( ! is_a( $user, WP_User::class ) || dokan_is_user_seller( $user->ID ) ) {
+            return;
+        }
+
+        $user_id       = $user->ID;
+        $current_roles = (array) $user->roles;
+
+        // Remove the customer role.
+        $user->remove_role( 'customer' );
+        if ( ! empty( $current_roles ) ) {
```

Comment:
Could you look into why we are deleting all the user roles of the user? As far as I know, one user can have multiple user roles. This might create problems if a user is assigned other roles via other plugins.